numba/tests/test_record_dtype.py

```diff
@@ -1553,6 +1566,22 @@ def test_shape(self):
         arr_res = cfunc(arg)
         np.testing.assert_equal(arr_res, arr_expected)
 
+    def test_corner_slice(self):
+        # testing corner cases while slicing nested arrays
+        nbarr = np.recarray((1, 2, 3, 5, 7, 13, 17), dtype=recordwith4darray)
+        for index, _ in np.ndenumerate(nbarr):
+            nbarr[index].p = np.random.randint(0, 1000, (3, 2, 5, 7),
```

Comment:
Can we seed the RNG in this test if we're using `np.random` please? (This just makes it easier when investigating any issues, so that the same numbers are always presented on every run instead of seeing variation from run to run) - e.g. adding `np.random.seed(1)` to the beginning of this test case.